R/lint.R

```diff
@@ -34,18 +37,37 @@ NULL
 #' }
 #'
 #' @export
-lint <- function(filename, linters = NULL, cache = FALSE, ..., parse_settings = TRUE) {
+lint <- function(filename, linters = NULL, cache = FALSE, ..., parse_settings = TRUE, text = NULL) {
 
-  inline_data <- rex::re_matches(filename, rex::rex(newline))
-  if (inline_data) {
-    content <- gsub("\n$", "", filename)
+  if (is.null(text)) {
+    inline_data <- rex::re_matches(filename, rex::rex(newline))
+    if (inline_data) {
+      text <- gsub("\n$", "", filename)
+      filename <- NULL
+    }
+  } else {
+    inline_data <- TRUE
+    if (length(text) > 1) {
+      text <- paste0(text, collapse = "\n")
+    }
+  }
+
+  no_filename <- missing(filename) || is.null(filename)
+
+  if (inline_data && no_filename) {
     filename <- tempfile()
-    on.exit(unlink(filename), add = TRUE)
```

Comment:
why remove `add=TRUE`? is there no other `on.exit`?

Reply:
Just my negligence, sorry, reverted to using `on.exit(add=TRUE)` as before.